ddtrace/tracer/sampler_test.go

```diff
@@ -1410,32 +1410,76 @@ func TestSamplingRuleMarshall(t *testing.T) {
 		in  SamplingRule
 		out string
 	}{
-		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), nil, 0, 0, nil, nil, SamplingRuleTrace, nil},
-			`{"service":"srv.[0-9]+","sample_rate":0,"type":"trace(1)"}`},
-		{SamplingRule{regexp.MustCompile("srv.*"), regexp.MustCompile("ops.[0-9]+"), 0, 0, nil, nil, SamplingRuleTrace, nil},
-			`{"service":"srv.*","name":"ops.[0-9]+","sample_rate":0,"type":"trace(1)"}`},
-		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), regexp.MustCompile("ops.[0-9]+"), 0.55, 0, nil, nil, SamplingRuleTrace, nil},
-			`{"service":"srv.[0-9]+","name":"ops.[0-9]+","sample_rate":0.55,"type":"trace(1)"}`},
-		{SamplingRule{nil, nil, 0.35, 0, regexp.MustCompile("http_get"), nil, SamplingRuleTrace, nil},
-			`{"resource":"http_get","sample_rate":0.35,"type":"trace(1)"}`},
-		{SamplingRule{nil, nil, 0.35, 0, regexp.MustCompile("http_get"), map[string]*regexp.Regexp{"host": regexp.MustCompile("hn-*")}, SamplingRuleTrace, nil},
-			`{"resource":"http_get","sample_rate":0.35,"tags":{"host":"hn-*"},"type":"trace(1)"}`},
-		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), regexp.MustCompile("ops.[0-9]+"), 0.55, 0, nil, nil, SamplingRuleSpan, nil},
-			`{"service":"srv.[0-9]+","name":"ops.[0-9]+","sample_rate":0.55,"type":"span(2)"}`},
-		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), regexp.MustCompile("ops.[0-9]+"), 0.55, 1000, nil, nil, SamplingRuleSpan, nil},
-			`{"service":"srv.[0-9]+","name":"ops.[0-9]+","sample_rate":0.55,"type":"span(2)","max_per_second":1000}`},
-		{SamplingRule{nil, nil, 1, 0, regexp.MustCompile("//bar"), nil, SamplingRuleTrace, nil},
-			`{"resource":"//bar","sample_rate":1,"type":"trace(1)"}`},
+		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), nil, 0, 0, nil, nil, SamplingRuleTrace, nil, nil},
+			`{"service":"srv.[0-9]+","sample_rate":0,"type":"1"}`},
+		{SamplingRule{regexp.MustCompile("srv.*"), regexp.MustCompile("ops.[0-9]+"), 0, 0, nil, nil, SamplingRuleTrace, nil, nil},
+			`{"service":"srv.*","name":"ops.[0-9]+","sample_rate":0,"type":"1"}`},
+		{SamplingRule{regexp.MustCompile("srv.[0-9]+"), regexp.MustCompile("ops.[0-9]+"), 0.55, 0, nil, nil, SamplingRuleTrace, nil, nil},
+			`{"service":"srv.[0-9]+","name":"ops.[0-9]+","sample_rate":0.55,"type":"1"}`},
```

Comment:
General comment: from what I understand, the PR should ensure that any used regex follows a "glob pattern" instead of the Go regex pattern. Both may be serialized to string in equal forms (like `srv.[0-9+`). I guess that's why I don't find any diff related to those.

Also, I understand that the type field was in an unexpected format too.

Apart from that, it looks good to me.

Reply:
> the PR should ensure that any used regex follows a "glob pattern

I can do it in a separate PR, as that would be a bit tedious.

Reply:
No, it's fine. I wanted to make sure I understood the task at hand.